WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift

```diff
@@ -103,10 +111,38 @@ class ReaderDetailViewController: UIViewController, ReaderDetailView {
         coordinator?.start()
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        configureFeaturedImage()
+
+        featuredImage.configure(scrollView: scrollView,
+                                navigationBar: navigationController?.navigationBar)
+
+        featuredImage.applyTransparentNavigationBarAppearance(to: navigationController?.navigationBar)
+
+        guard !featuredImage.isLoaded else {
+            return
+        }
+
+        // Load the image
+        featuredImage.load { [unowned self] in
+            self.hideLoading()
```

Comment:
I have a big concern about that here: waiting for the featured image to load to display the content might be ok in good connections, but almost impracticable in 3G for example.

I tested a few articles here (in iPad and iPhone using 3G) and for some articles it took so long that I give up (loading a GIF in that case). You can see my [explorations here](https://d.pr/v/AAQimY).

I know changing that probably opens a lot of other issues. But I really think the most important thing is the content, and once available we should show it right away. We can add a grey placeholder for the featured image once it's being loaded.

Reply:
I agree, I have a plan to address this but I'll do it in a separate PR.